src/cache.rs

```diff
@@ -142,22 +168,45 @@ impl Cache {
         }
     }
 
+    fn find_custom_patch(name: &str, custom_dir: &Path) -> Option<PathBuf> {
+        // Look up custom patch (<name>.patch). If it exists, store it in a variable.
```

Comment:
I'm generally a fan of many comments, because they help summarizing the logic below. But it could be moved to the method docs:

```
/// Look up custom patch (<name>.patch). If it exists in the file system, return the file path.
fn find_custom_patch(name: &str, custom_dir: &Path) -> Option<PathBuf> {
```